internal/pkg/skuba/cni/cilium.go

```diff
@@ -145,6 +187,98 @@ func CreateOrUpdateCiliumConfigMap(client clientset.Interface) error {
 	return nil
 }
 
+// MigrateEtcdToCrd performs the migration of Cilium internal data from etcd
+// cluster to CRD during upgrade from Cilium 1.5 to Cilium 1.6. This step is not
+// mandatory, without it, Cilium is going to regenerate data from scratch which
+// might result in service downtimes. If the automated migration is not
+// successful, the upgrade should still proceed with the risk of temporary
+// service downtimes.
+func MigrateEtcdToCrd(client clientset.Interface, config *rest.Config) error {
+	// Find any Cilium pod.
+	var ciliumPod string
+	if err := retry.OnError(retry.DefaultRetry, IsErrCiliumNotFound, func() error {
+		pods, err := client.CoreV1().Pods(metav1.NamespaceSystem).List(metav1.ListOptions{})
```

Comment:
Q: can't the client filter more deeply, to avoid us to do this filtering ourselves? I am still learning, and I don't know its capabilities yet.

Reply:
Now I see that `ListOptions` as an attribute `LabelSelector`, I can try to use it.

Reply:
It looks like it's gonna reduce the code on our side, which is I think better (I expect more scrutiny of the client-go than skuba)